### PR TITLE
feat: Core service cdk deployment

### PR DESCRIPTION
**Pull request by chejimmy**
# Description

Added cdk package to deploy Core service and dependent resources
Pending testings and updating IAM policies to have the least-privilege permissions

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Deployment succeeded

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: cdk/lib/iot-application-stack.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    new CoreStack(this, 'Core', {
+      coreServiceProps: {
+        databaseTableArn: databaseStack.resourceTable.tableArn,
+        databaseTableName: databaseStack.resourceTable.tableName,
+        userPoolClientId: authStack.userPoolClient.userPoolClientId,
+        userPoolId: authStack.userPool.userPoolId,
+      },
+    });
+  }
+}
```
Comment (tracy-french) [inline cdk/lib/iot-application-stack.ts:22]: Clean! Love it!

File: cdk/lib/core/core-service.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  constructor(scope: Construct, id: string, props: CoreServiceProps) {
+    super(scope, id);
+
+    ({
+      databaseTableArn: this.databaseTableArn,
+      databaseTableName: this.databaseTableName,
+      userPoolClientId: this.userPoolClientId,
+      userPoolId: this.userPoolId,
+    } = props);
```
Comment (tracy-french) [inline cdk/lib/core/core-service.ts:33]: Nit: does this need parens?

**Review by tracy-french — APPROVED** [review]
Awesome work! Your change is pretty much perfect! Well done, Jimmy! :)

One thing I am a little conflicted on is having business rules mixed with the implementation logic, but I guess that's how it's supposed to be being IaC lol.

**Review by tracy-french — APPROVED** [review]

File: cdk/lib/core/core-service.ts (hunk 2 above)
Comment (chejimmy) [reply cdk/lib/core/core-service.ts:33]: The parentheses are required, from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment):

> The parentheses ( ... ) around the assignment statement are required when using object literal destructuring assignment without a declaration.

the other way to do this is to first destructure into const vars and then assign to fields:

```
  const {
    databaseTableArn,
    databaseTableName,
    ...
  } = props;

  this.databaseTableArn = databaseTableArn;
  this.databaseTableName = databaseTableName;
  ...
```

but I prefer doing the destructuring assignment so it skipped the unnecessary const vars assignments

**Pull request merged**